File: Тренировка 1.0/01. Сложность, тестирование, особые случаи/H. Метро/Метро.cs
using System;

int ai = int.Parse(Console.ReadLine());
int bi = int.Parse(Console.ReadLine());
int an = int.Parse(Console.ReadLine());
int bn = int.Parse(Console.ReadLine());

int minATime = an + (an - 1) * ai;
int maxATime = an + (an + 1) * ai;
int minBTime = bn + (bn - 1) * bi;
int maxBTime = bn + (bn + 1) * bi;

int min = Math.Max(minATime, minBTime);
int max = Math.Min(maxATime, maxBTime);

if(min > max) Console.WriteLine("-1");
else Console.WriteLine(min + " " + max);

File: Тренировка 1.0/01. Сложность, тестирование, особые случаи/H. Метро/Метро.cpp
#include <iostream>
#include <cmath>

int main()
{
    int ai, bi, an, bn;
    std::cin >> ai >> bi >> an >> bn;

    int minATime = an + (an - 1) * ai;
    int maxATime = an + (an + 1) * ai;
    int minBTime = bn + (bn - 1) * bi;
    int maxBTime = bn + (bn + 1) * bi;

    int min = std::max(minATime, minBTime);
    int max = std::min(maxATime, maxBTime);

    if (min > max) std::cout << "-1";
    else std::cout << min << " " << max;
}
